File: sdk-python/examples/contacts/retrieve.py
"""Run with: python examples/contacts/retrieve.py"""

from __future__ import annotations

from threecommon import ThreeCommon


def main() -> None:
    with ThreeCommon(api_key="3co_your_api_key_here") as client:
        contact = client.contacts.retrieve("cnt_replace_with_real_id")
        print(f"{contact.full_name} <{contact.email}>")
        print(f"  status:     {contact.status}")
        print(f"  orders:     {contact.order_sum}")
        print(f"  gross:      {contact.gross_sum}")
        print(f"  vendor_id:  {contact.vendor_id}")


if __name__ == "__main__":
    main()

File: sdk-node/examples/contacts/retrieve.ts
/**
 * Retrieve a single contact by id.
 *
 * Run:
 *   npx tsx examples/contacts/retrieve.ts
 */

import { ThreeCommon } from '@3common/sdk'

const client = new ThreeCommon({
  apiKey: '3co_your_api_key_here',
})

const contact = await client.contacts.retrieve('cnt_replace_with_real_id')

console.log(`${contact.fullName} <${contact.email}>`)
console.log(`  status:      ${contact.status}`)
console.log(`  orders:      ${String(contact.orderSum)}`)
console.log(`  gross:       ${String(contact.grossSum)}`)
console.log(`  vendorId:    ${contact.vendorId}`)
